Detail page/review page

sunghyun1356 opened the pull request:
모델수정

sunghyun1356 committed:
model 수정 후 upload

sunghyun1356 committed:
null=True 수정

sunghyun1356 committed:
null=True 수정

sunghyun1356 committed:
Restaurants에 reservation 속성 추가

sunghyun1356 committed:
model 수정

sunghyun1356 committed:
restaurant 기본 정보 api 구현 및 여러 모델 url 수정 (#1)

sunghyun1356 committed:
requirements.txt 파일 수정 -> geopy설정

sunghyun1356 committed:
model 수정

sunghyun1356 committed:
레스토랑 상세 페이지 api구현 완료(#1)

sunghyun1356 committed:
레스토랑 상세 페이지 api구현 완료(#1)

sunghyun1356 committed:
레스토랑 디테일 페이지 API 수정 -> 유저 is_staff=True로 변경(#1)

go-the-extra-mile commented on Restaurants/models.py:
```diff
@@ -1,3 +1,63 @@
+
+class Category(models.Model):
+    name = models.CharField(max_length=100, verbose_name="category_name")
+    image = models.ImageField(upload_to="", default='media/default_image.jpeg')
```
미디어 업로드 관련한 세팅이 아직 없는 것 같습니다! 이대로면 업로드가 이루어지지 않거나 프로젝트 디렉터리 자체에 업로드가 되어버릴 것 같습니다. 
[참고링크](https://stackoverflow.com/questions/34563454/django-imagefield-upload-to-path)

sunghyun1356 replied:
이거 postman에서 오류가 나서 아직 설정을 안했어여 뮤ㅠㅠㅠ

go-the-extra-mile commented on Papago_API.py:
```diff
@@ -0,0 +1,26 @@
+    print(response_body.decode('utf-8'))
+else:
+    print("Error Code:" + rescode)
+
```
def translate(lang='en', text=None) -> 문자열 형식의 lang 언어로 번역된 text, bool 타입의 성공여부
이런 형식을 가지는 하나의 독립적인 함수로 만들어주시면 좋을 것 같습니다!

sunghyun1356 replied:
넵 수정하겠ㅅ브니다

sunghyun1356 committed:
detail_review페이지 리뷰 정렬(latest,rate별) 구현 (#4)

sunghyun1356 committed:
detail_review페이지 리뷰 정렬(latest,rate별) 구현 (#4)

go-the-extra-mile commented on Restaurants/models.py:
```diff
@@ -1,3 +1,63 @@
+class Restaurant(models.Model):
+    name = models.CharField(max_length=100, verbose_name="restaurant_name", null=False)
+    address = models.CharField(max_length=200, verbose_name="restaurant_address", null=False)
+    open_close_time = models.TextField(max_length=1000, verbose_name="open_close_time", null=True)
```
```
open_time = models.TimeField()
close_time = models.TimeField()
```
로 변경하면 프론트에서 원하는 형태로 가공해서 전달하기 좋을 것 같습니다

sunghyun1356 replied:
그러면 월~일 까지의 시간대가 필요할텐데 따로 다 각각 넘겨주는게 편하지 않을까염?

go-the-extra-mile commented on Restaurants/models.py:
```diff
@@ -1,3 +1,63 @@
+    latitude = models.FloatField(max_length=100, null=False)
+    longitude= models.FloatField(max_length=100, null=False)
+    koogle_ranking = models.IntegerField(verbose_name="koogle_ranking")
+    reservation = models.BooleanField(verbose_name="reservation", default=False)
```
reservation_link가 NULL이면 예약링크가 없다는 것을 이미 내포한다고 이해할 수 있습니다. 그러면 이 필드와 clean()이 없어도 될 것 같습니다.

sunghyun1356 replied:
이게 코드에 null=True라고 지정했는데도 계속 링크를 넣으라고 admin사이트에서 오류를 내보내서 적은 건데 나중에 수정해볼게요(똑같은 에러로 이미지파일처리도 아직 안한것!)

sunghyun1356 committed:
likes data view 수정 (#1)

sunghyun1356 committed:
menu->content 삭제(#1)

go-the-extra-mile commented on Reviews/models.py:
```diff
@@ -1,3 +1,42 @@
 # Create your models here.
+
+class Review(models.Model):
+    user = models.ForeignKey(User, on_delete=models.CASCADE)
```
models.py에서 사용자 모델을 참조할 때는 재사용성을 위해 settings.AUTH_USER_MODEL을 참조하는게 더 좋을 것 같습니다. 
[참고](https://docs.djangoproject.com/en/4.2/topics/auth/customizing/#referencing-the-user-model)

sunghyun1356 replied:
넵 수정하겠ㅅ브니다

go-the-extra-mile commented on Reviews/models.py:
```diff
@@ -1,3 +1,42 @@
+    image_1 = models.ImageField(upload_to="",null=True)
+    image_2 = models.ImageField(upload_to="",null=True)
+    image_3 = models.ImageField(upload_to="", null=True)
+    created_at = models.DateTimeField(auto_created=True)
```
auto_now_add를 의미하신 것 같습니다

sunghyun1356 replied:
넵 수정하겠습니다

go-the-extra-mile commented on Reviews/models.py:
```diff
@@ -1,3 +1,42 @@
+        likes_type = self.likes.likes if self.likes else "Unknown Likes Type"
+        return f"Review: {restaurant_name}'s {likes_type} by {self.review.user.username}"
+    
+class Review_Restaurant(models.Model):
```
likes와 restaurant을 다대다로 연결하는 Likes_Restaurant이 더 적합할 것 같습니다. 
하나의 리뷰가 여러 레스토랑에 대해 쓰이지는 않습니다.

sunghyun1356 replied:
이부분 순환에러가 떠서 일단은 이러헤 해놓았는데 admin상과 postman상에서는 문제가 없어서 남겨놓았는데 review_create까지 구현해보고 수정하겠습니다(데이터를 다시 넣어야해서요 ㅠ)

go-the-extra-mile reviewed (commented):
모델만 리뷰했습니다. 제가 언급한 내용으로 수정한 커밋을 날리겠습니다. 수정한 파일 확인해주시면 감사하겠습니다.

go-the-extra-mile committed:
edit Restaurant, Review app's models to accomodate review on PR

sunghyun1356 committed:
만들어진 날짜 보이게 하기 수정(#4)

sunghyun1356 committed:
만들어진 날짜 보이게 하기 수정(#4)

go-the-extra-mile committed:
Restaurants, Reviews 앱의 모델 수정

- 정상 동작하도록 Restaurants, Reviews 앱의 모델 수정
- 식당 영업시간 저장하는 방법으로 OpenHours 모델 도입
- 로컬 세팅과 배포 세팅 분리
    - 배포 세팅으로 서버를 켜려면 `python manage.py runserver --settings=config.settings.production`이라고 쓰시면 됩니다

sunghyun1356 committed:
update

sunghyun1356 committed:
Merge branch 'main' of https://github.com/hackerton4-koogle/server into Detail-Page/Review-Page

sunghyun1356 committed:
model이랑 settings 수정요청건 완료

sunghyun1356 committed:
settings, model 수정 완료

sunghyun1356 committed:
settings, model 수정 완료

sunghyun1356 committed:
review쓰기 구현(#4)

sunghyun1356 committed:
리뷰작성하기 구현(#4)

sunghyun1356 committed:
번역 API구현/Papago_API는 구현된 함수/Translate는 외부 파일에서 실행 가능한 예시 파일(#3)

sunghyun1356 committed:
현서 코드 합치기

sunghyun1356 committed:
지금까지 모두다 합치기

sunghyun1356 committed:
user url추가
